### PR TITLE
testing entity's methods to cover these classes by unittests 

### PR DESCRIPTION
**Pull request by dshulgin2015**

### CODE REVIEW

**Comment by dshulgin2015** [comment]
create new pull request to whole team account

File: src/evrythng/tests/config.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{    "parameters": {
+        "API_KEY": "FSz26K5a4HlImg7hX4i1BhEE6B5JieAKDJKijzAdX2a3S8AjyF1H406qbcIxT6vHTlqe2pORLc9Ts8TJ"
```
Comment (rtrajano) [inline src/evrythng/tests/config.json]: It's generally not a good idea to check in keys into the repo. This should be pulled in as an environment variable.

**Comment by rtrajano** [comment]
Also, if you could set up your IDE to auto pep8/pycodestyle and pylint/pyflakes your code, it will catch a bunch of issues. You have some unused imports as well as simple code style issues in this PR.

https://pypi.python.org/pypi/flake8 will do both syntax check and style checks (a combination of pep8 and pyflakes)

File: src/evrythng/tests/config.json (hunk 1 above)
Comment (dshulgin2015) [reply src/evrythng/tests/config.json]: right, i was thinking about this.

**Comment by dshulgin2015** [comment]
I have unused imports because i'm going to write unittests for all entities, so i will be used.

**Comment by dshulgin2015** [comment]
checkout my last commits

File: src/evrythng/tests/test_entities.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+        count = len(products.list_products().json())
+        product_id = products.create_product(
+            name='test_creating_prod_GooeeIOT').json()['id']
+        count += 1
+        status_code = products.delete_product(
+            product_id=str(product_id)).status_code
+        self.assertEqual(status_code, 200)
+        self.assertEqual(len(products.list_products(
+            api_key=API_KEY).json()), count - 1)
```
Comment (rtrajano) [inline src/evrythng/tests/test_entities.py]: `API_KEY` is no longer defined in the file. Pyflakes would have caught this.

Comment (dshulgin2015) [reply src/evrythng/tests/test_entities.py]: done.

**Comment by rtrajano** [comment]
Nice! **CR+1** 👍

File: src/evrythng/tests/test_entities.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    list_products
+    read_product +
+    update_product
+    delete_product
+
+    on a test EVT account by Dmitry Shulgin.
+
+    '''
+
+    def test1_creation(self):
```
Comment (ramonPaulAlvarez) [inline src/evrythng/tests/test_entities.py]: You could 'test' the creation in a setUp method and 'test' deletion in tearDown method.  It'd be less code, but the same number of calls.  It really helps to reduce test size when dealing with common setup/teardown code.

Comment (dshulgin2015) [reply src/evrythng/tests/test_entities.py]: Ohh, i see. 
http://stackoverflow.com/questions/6854658/explain-the-setup-and-teardown-methods-used-in-test-cases

Comment (dshulgin2015) [reply src/evrythng/tests/test_entities.py]: will fix it tomorrow.

**Comment by dshulgin2015** [comment]
looks better now. Please check it out.

File: src/evrythng/tests/test_entities.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+from evrythng.entities import actions
+from evrythng.entities import products
+
+
+class TestProductMethods(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.product_id = products.create_product(
```
Comment (ramonPaulAlvarez) [inline src/evrythng/tests/test_entities.py:15]: Will you please add """Product Tests Setup""" here and then the other missing doc strings?  The """ is more consistent with the rest of our code base too.

Check [this](https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings) out for information on doc string conventions.  Though ''' is in fact needed when you have a """ within the doc string.

File: src/evrythng/tests/test_entities.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+            name='test_creating_prod_GooeeIOT').json()['id']
+
+    @classmethod
+    def tearDownClass(cls):
+        products.delete_product(
+            product_id=str(cls.product_id))
+
+    def test1_creation(self):
+        ''' creating test product and check if existing name is equal to test str'''
+
```
Comment (ramonPaulAlvarez) [inline src/evrythng/tests/test_entities.py:28]: > There's no blank line either before or after the docstring.

**Comment by ramonPaulAlvarez** [comment]
Excellent updates!

**Comment by dshulgin2015** [comment]
fixed

**Comment by dshulgin2015** [comment]
I'm going to create read_action method this weekend and more tests in this script.

File: src/evrythng/entities/actions.py [hunk 6]
```diff
@@ -47,8 +46,16 @@ def delete_action(type_, action_id, api_key=None):
...
+    url = '/actions/{}/{}'.format(type_, action_id)
+    return utils.request('GET', url, api_key=api_key)
+
+
 def list_actions(type_, api_key=None, **request_kwargs):
     """List Actions"""
     assertions.datatype_str('type_', type_)
     url = '/actions/{}'.format(type_)
-    return utils.request('GET', url, api_key=api_key, **request_kwargs)
+    return utils.request('GET', url, api_key=api_key, **request_kwargs)
```
Comment (augustawind) [inline src/evrythng/entities/actions.py]: make sure all your files end with a newline or bad stuff can happen!

File: src/evrythng/tests/test_entities.py [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+    @classmethod
+    def tearDownClass(cls):
+        """Product Tests Destroying"""
+
+        products.delete_product(
+            product_id=str(cls.product_id), api_key=API_KEY)
+
+    def test1_creation(self):
+        """creating test product and
+         check if existing name is equal to test str"""
```
Comment (augustawind) [inline src/evrythng/tests/test_entities.py:27]: Check out the official PEP for docstrings: https://www.python.org/dev/peps/pep-0257/
That's our standard for docstrings :)

File: src/evrythng/tests/test_entities.py [hunk 8]
```diff
@@ -0,0 +1,82 @@
+import unittest
+import os
+from evrythng.entities import actions
+from evrythng.entities import products
```
Comment (augustawind) [inline src/evrythng/tests/test_entities.py:5]: Check out the imports section in PEP 8: https://www.python.org/dev/peps/pep-0008/#imports

(The only thing missing here is a blank line between stdlib imports and local imports)

File: src/evrythng/tests/test_entities.py [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+            name='test_creating_prod_GooeeIOT', api_key=API_KEY).json()['id']
+
+    @classmethod
+    def tearDownClass(cls):
+        """Product Tests Destroying"""
+
+        products.delete_product(
+            product_id=str(cls.product_id), api_key=API_KEY)
+
+    def test1_creation(self):
```
Comment (codeadict) [inline src/evrythng/tests/test_entities.py]: I personally don't like numbering tests like test1 test2 but thats just naming, overall looks good to me.

Comment (dshulgin2015) [reply src/evrythng/tests/test_entities.py]: i named it first time to run it circle. will fix it.

File: src/evrythng/tests/test_entities.py (hunk 7 above)
Comment (dshulgin2015) [reply src/evrythng/tests/test_entities.py:27]: thx, just did this docstr with pep8 plugin for subl.

File: src/evrythng/tests/test_entities.py (hunk 9 above)
Comment (dshulgin2015) [reply src/evrythng/tests/test_entities.py]: fixed

File: src/evrythng/tests/test_entities.py (hunk 8 above)
Comment (dshulgin2015) [reply src/evrythng/tests/test_entities.py:5]: fixed

File: src/evrythng/tests/test_entities.py (hunk 7 above)
Comment (dshulgin2015) [reply src/evrythng/tests/test_entities.py:27]: fixed

File: src/evrythng/entities/actions.py (hunk 6 above)
Comment (dshulgin2015) [reply src/evrythng/entities/actions.py]: fixed

File: src/evrythng/entities/products.py [hunk 10]
```diff
@@ -59,3 +59,9 @@ def delete_product(product_id, api_key=None):
     assertions.datatype_str('product_id', product_id)
     url = '/products/{}'.format(product_id)
     return utils.request('DELETE', url, api_key=api_key)
+
+def read_all_actions(product_id, action_type='all', api_key=None):
```
Comment (ramonPaulAlvarez) [inline src/evrythng/entities/products.py:63]: - Will you add a docstring for this and the other read_all_actions?
- I recommend adding in the parameters too.  Reference create_collection for guidelines.
- Perhaps action_type default should be None and you can use 'all' when None is provided?

File: src/evrythng/tests/test_entities.py [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+
+    @classmethod
+    def tearDownClass(cls):
+        """Product Tests Destroying"""
+
+        products.delete_product(
+            product_id=str(cls.product_id), api_key=API_KEY)
+
+    def test_creation(self):
+        """creating test product and
```
Comment (ramonPaulAlvarez) [inline src/evrythng/tests/test_entities.py:26]: I think the first sentence of a docstring should fit on one line with respect to max line width.  I usually use a summary for the first line and if need be I add more below it on a new line.

**Comment by LyleScott** [comment]
What's the status of this? Can we get this merged in?

**Pull request closed without merging**